dsl_lang/lib/parser.mly

```diff
@@ -3,29 +3,33 @@
 
     exception FormatError of string
 
+    (* error messages *)
+    let misplaced_return = "Program must contain exactly one return statement as its final component."
+    let misplaced_declare = "Cannot interleave declarations and assignments."
+    let missing_declare = "Program must begin with a declaration of classes."
+
     type internalcomp =
     | DeclareComp of clss list
     | AssnComp of var * policy
     | RtnComp of policy
 
     (* Check that a program contains exactly one return statement as its final component *)
     let rec validate_seq acc = function
-    | RtnComp(pol) :: [] -> acc, pol
-    | RtnComp(_) :: _ ->
-        raise(FormatError "Program must contain exactly one return statement as its final component.")
+    | RtnComp pol :: [] -> acc, pol
+    | RtnComp _ :: _ -> raise (FormatError misplaced_return)
     | h :: t -> validate_seq (acc @ [h]) t
-    | [] -> raise (FormatError "Program must begin with a declaration of classes.")
+    | [] -> raise (FormatError misplaced_return)
```

Comment:
Minor typo fix: the last commit was for this line. Prior to [6bab7b9](https://github.com/cucapra/packet-scheduling/pull/39/commits/6bab7b96f66a94439006389e749efee029e81363), if we parsed this program
```
classes A, B;

policy = A;
```
the parser would say "Program must begin with a declaration of classes." when we wanted it to say "Program must contain exactly one return statement as its final component."

Reply:
Thanks – good catch!